microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -0,0 +1,421 @@
+
+1. Below the `MainPage` constructor, add the following `InitializeWebView2Async` method:
+
+   ```csharp
+   private async void InitializeWebView2Async()
+   {
+      await WebView2.EnsureCoreWebView2Async();
+      var dispatchAdapter = new WinRTAdapter.DispatchAdapter();
+      WebView2.CoreWebView2.AddHostObjectToScript("Windows", dispatchAdapter.WrapNamedObject("Windows", dispatchAdapter));
+   }
+   ```
+
+   This method calls `AddHostObjectToScript`.
+
+1. In the `MainPage` constructor, above the `StatusUpdate("Ready");` line, add the following code:
+
+   ```csharp
+   InitializeWebView2Async();
+   ```
+
+1. Right-click the **webview2_sample_uwp** project, and then select **Set as startup project**.
+
+1. Press `Shift+Ctrl+S` to save all files.
+
+1. Press `F5` to run the sample app.  The **WebView2 WinUI 2 UWP Sample** window opens.
+
+   ![The WebView2 WinUI 2 UWP Sample window.](winrt-from-js-images/webview2-winui-2-uwp-sample-app-window.png)
+
+The host app's web-side code (and the DevTools Console) can now call methods and properties of the specified namespaces or classes of the host object.
+
+
+<!-- =============================================== -->
+## Step 8. Call methods and properties on the host object from web-side JavaScript
+
+Next, use the DevTools Console to demonstrate that web-side code can call the included, specified host-side APIs.
+
+1. Click in the main part of the WebView2 sample app window to give it focus, and then press `Ctrl+Shift+I` to open Microsoft Edge DevTools.  Or, right-click the page, and then select **Inspect**.  The Microsoft Edge DevTools window opens.
+
+1. If the Microsoft Edge DevTools window isn't visible, press `Alt+Tab` to display the DevTools window.  If needed, move the DevTools window.
+
+1. In the DevTools Console, paste the following code, and then press `Enter`:
+
+   ```javascript
+   const Windows = chrome.webview.hostObjects.sync.Windows;
+   (new Windows.Globalization.Language("en-US")).displayName;
+   ```
+
+   The Console outputs a language name string, such as `English (United States)`, demonstrating that your app's host-side code can be called from web-side code:
+
+   ![Using the DevTools Console to test calling native-side code from web-side code.](winrt-from-js-images/devtools-console-calling-native-side-code.png)
+
+Congratulations!  You've finished the sample demonstration of calling WinRT code from JavaScript code.
+
+
+<!-- ====================================================================== -->
+## Make AddHostObjectToScript act like Chakra WinRT projection
+
+The `AddHostObjectToScript` feature defaults to using asynchronous and verbose proxies.  You can read more about that in the `AddHostObjectToScript` documentation; use the [See also](#see-also) links below.  To make `AddHostObjectToScript` act more like the Chakra WinRT projection, set the following properties:
```

Comment:
It took me a while to make sense out of the second sentence and realize how to get to the docs. Would something like the below be clearer?

> To read more about `AddHostObjectToScript` and its default behavior, use the [See also](#see-also) links below.